python_hll/serialization.py

```diff
@@ -622,7 +622,7 @@ def pack_version_byte(cls, schema_version, type_ordinal):
         :returns: the packed version byte
         :rtype: byte
         """
-        return BitUtil.to_signed_byte(BitUtil.left_shift_int(cls.NIBBLE_MASK & schema_version, cls.NIBBLE_BITS) | (cls.NIBBLE_MASK & type_ordinal))
+        return BitUtil.left_shift_int(cls.NIBBLE_MASK & schema_version, cls.NIBBLE_BITS) | (cls.NIBBLE_MASK & type_ordinal)
```

Comment:
The corresponding Java code explicitly casts to a byte, which is why we do it here:

https://github.com/aggregateknowledge/java-hll/blob/1f4126e79a85b79581460c75bf1c1634b8a7402d/src/main/java/net/agkn/hll/serialization/SerializationUtil.java#L154